### PR TITLE
Refactor entering degrees spec

### PR DESCRIPTION
**Pull request by TobyRet**
## Context

The `candidate_entering_degrees` system spec has grown quite long. 
It's probably a good candidate for refactoring

## Changes proposed in this pull request

- Split spec into smaller specs

## Guidance to review

- Will this make life easier?
- Happy with the new folder structure or should we flatten the structure as before?

## Link to Trello card

https://trello.com/c/g3MEVPhh/2668-medium-refactor-candidateenteringdegrees-system-spec

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Review by davidgisbey — APPROVED** [review]
lgtm!

**Review by stevehook — APPROVED** [review]
:+1: This looks better to me. I was a bit unsure originally because system specs are 'meant to be' long (to test a feature end-to-end), but this one was _too_ long and I think this is as good a way to divide the functionality as any I can think of.

File: spec/system/candidate_interface/entering_details/degrees/candidate_delete_and_replace_degrees_spec.rb [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+RSpec.feature 'Deleting and replacing a degree' do
+  include CandidateHelper
+
+  scenario 'Candidate deletes and replaces their degree' do
+    given_i_am_signed_in
+    and_i_have_completed_the_degree_section
+    and_i_visit_the_site
+    when_i_click_on_degree
+    then_i_can_check_my_undergraduate_degree
```
Comment (malcolmbaig) [inline spec/system/candidate_interface/entering_details/degrees/candidate_delete_and_replace_degrees_spec.rb:11]: Suggestion: collapse the above 3 into `when_i_view_the_degree_section`

File: spec/system/candidate_interface/entering_details/degrees/candidate_delete_and_replace_degrees_spec.rb [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+    when_i_add_my_degree_back_in
+    and_i_click_on_continue
+    then_i_should_see_the_form_and_the_section_is_not_completed
+    when_i_click_on_degree
+    then_i_can_check_my_undergraduate_degree
+
+    when_i_click_on_add_another_degree
+    then_i_see_the_add_another_degree_form
+    when_i_fill_in_my_additional_degree
```
Comment (malcolmbaig) [inline spec/system/candidate_interface/entering_details/degrees/candidate_delete_and_replace_degrees_spec.rb:26]: Suggestion: collapse the above 3 into `when_i_add_another_degree`

File: spec/system/candidate_interface/entering_details/degrees/candidate_editing_degrees_spec.rb [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+RSpec.feature 'Editing a degree' do
+  include CandidateHelper
+
+  scenario 'Candidate edits their degree' do
+    given_i_am_signed_in
+    and_i_have_completed_the_degree_section
+    and_i_visit_the_site
+    when_i_click_on_degree
+    then_i_can_check_my_undergraduate_degree
```
Comment (malcolmbaig) [inline spec/system/candidate_interface/entering_details/degrees/candidate_editing_degrees_spec.rb:11]: Suggestion: collapse the above 3 into `when_i_view_the_degree_section`

File: spec/system/candidate_interface/entering_details/degrees/candidate_entering_degrees_spec.rb [hunk 4]
```diff
@@ -0,0 +1,203 @@
+require 'rails_helper'
+
+RSpec.feature 'Entering a degree' do
+  include CandidateHelper
+
+  scenario 'Candidate enters their degree' do
+    given_i_am_signed_in
+    and_i_visit_the_site
+    when_i_click_on_degree
+    then_i_see_the_undergraduate_degree_form
```
Comment (malcolmbaig) [inline spec/system/candidate_interface/entering_details/degrees/candidate_entering_degrees_spec.rb:10]: Suggestion: as above, collapse the above 3 into `when_i_view_the_degree_section`

File: spec/system/candidate_interface/entering_details/degrees/candidate_entering_degrees_spec.rb [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    then_i_see_validation_errors_for_degree_grade
+    when_i_select_the_degree_grade
+    and_i_click_on_save_and_continue
+
+    # Add years
+    then_i_can_see_the_start_and_graduation_year_page
+    when_i_click_on_save_and_continue
+    then_i_see_validation_errors_for_graduation_year
+    when_i_fill_in_the_start_and_graduation_year
+    and_i_click_on_save_and_continue
```
Comment (malcolmbaig) [inline spec/system/candidate_interface/entering_details/degrees/candidate_entering_degrees_spec.rb:48]: I'd be tempted to collapse these 'Add steps' into single methods, removing the need for the comments, eg -

`when_i_enter_the_degree_type`
`and_i_enter_the_subject`
`and_i_enter_the_institution`

And so on. There's an aversion in some projects to adding assertions to "when" steps, but I think it's fine if their purpose is to check you're on the right page or you're seeing an appropriate error message while carrying out some kind of action. Often leads to more helpful error messages when specs break, too.

**Review by malcolmbaig — APPROVED** [review]
Some suggestions for brevity, but on the whole LG 👍🏽

**Pull request merged**